Add KGO Dataplane documentation (#1384)
Kong/developer.konghq.com: app/operator/konnect/kongpluginbinding.md

* `Service` and `Route`
* `Service` and `Consumer`
* `Service` and `ConsumerGroup`
* `Service`, `Route`, and `Consumer`
* `Service`, `Route`, and `ConsumerGroup`
* `Consumer` and `ConsumerGroup`

For example, we can configure a `rate-limiting` plugin to a service and a consumer like this:

Create a service:

```shell
echo '
kind: KongService
apiVersion: configuration.konghq.com/v1alpha1
metadata:
  namespace: default
  name: service-plugin-binding-combination
spec:
  host: example.com
  controlPlaneRef:
    type: konnectNamespacedRef
    konnectNamespacedRef:
      name: cp
' | kubectl apply -f -
```

Create a consumer:

```shell
echo '
kind: KongConsumer
apiVersion: configuration.konghq.com/v1
metadata:
  namespace: default
  name: consumer-plugin-binding-combination
username: consumer-test
spec:
  controlPlaneRef:
    type: konnectNamespacedRef
    konnectNamespacedRef:
      name: cp
' | kubectl apply -f -
```

Create a plugin:

```shell
echo '
kind: KongPlugin
apiVersion: configuration.konghq.com/v1
metadata:
  namespace: default
  name: rate-limiting-minute-10
plugin: rate-limiting
config:
  policy: local
  minute: 10
' | kubectl apply -f -
```

> Then, you can create a `KongPluginBinding` including both references to the `KongService` and the `KongCosumer` to attach the plugin to the service and the consumer:

Reviewer: ```suggestion
Then, you can create a `KongPluginBinding` including both references to the `KongService` and the `KongCosumer` to attach the plugin to the Service and the Consumer:
```

```shell
echo '
kind: KongPluginBinding
apiVersion: configuration.konghq.com/v1alpha1
metadata:
  namespace: default
  name: binding-combination-service-consumer
spec:
  pluginRef:
    kind: KongPlugin
    name: rate-limiting-minute-10
  targets:
    serviceRef:
      group: configuration.konghq.com
      kind: KongService
      name: service-plugin-binding-combination
    consumerRef:
      name: consumer-plugin-binding-combination
  controlPlaneRef:
    type: konnectNamespacedRef
    konnectNamespacedRef:
      name: cp
' | kubectl apply -f -
```

### Attaching plugins globally to a control plane {% new_in 1.5 %}

> You can also attach a plugin globally to a control plane by setting the `spec.scope` field to `GlobalInControlPlane` in the `KongPluginBinding` CRD.

Reviewer: ```suggestion
You can also attach a plugin globally to a Control Plane by setting the `spec.scope` field to `GlobalInControlPlane` in the `KongPluginBinding` CRD.
```

> Create a `KongPluginBinding` to attach a plugin globally to a control plane like this:

Reviewer: ```suggestion
Create a `KongPluginBinding` to attach a plugin globally to a Control Plane like this:
```

```shell
echo '
kind: KongPluginBinding
apiVersion: configuration.konghq.com/v1alpha1
metadata:
  namespace: default
  name: binding-global-rate-limiting
spec:
  # This indicates that the plugin is attached globally to the control plane and allows leaving targets empty.
  scope: GlobalInControlPlane
  pluginRef:
    kind: KongPlugin
    name: rate-limiting-minute-10
  controlPlaneRef:
    type: konnectNamespacedRef
    konnectNamespacedRef:
      name: cp
' | kubectl apply -f -
```

> Having the `KongPluginBinding` created, the plugin will be attached globally to the control plane in {{ site.konnect_short_name }}.

Reviewer: ```suggestion
Having the `KongPluginBinding` created, the plugin will be attached globally to the Control Plane in {{ site.konnect_short_name }}.
```

## Using annotations to bind plugins to other entities

> {:.note}

Reviewer: ```suggestion
{:.info}
```

> > **NOTE:** This approach is considered legacy and using `KongPluginBinding` CRD is recommended instead.

Reviewer: ```suggestion
> **Note:** This approach is considered legacy and using `KongPluginBinding` CRD is recommended instead.
```

> Users can expect that `konghq.com/plugins` annotation support will be removed at some point in the future.

You can also use the `konghq.com/plugins` annotation to attach plugins to other entities like it's done in {{ site.kic_product_name }}.
The {{ site.kgo_product_name }} will create `KongPluginBinding` resources for the annotations and configure them in {{ site.konnect_short_name }}.

In the example above, you can create a `KongPlugin` and a `KongService` like this:

```shell
echo '
kind: KongPlugin
apiVersion: configuration.konghq.com/v1
metadata:
  namespace: default
  name: rate-limiting-minute-10
plugin: rate-limiting
config:
  policy: local
  minute: 10
' | kubectl apply -f - 
```

```shell
echo '
kind: KongService
apiVersion: configuration.konghq.com/v1alpha1
metadata:
  namespace: default
  name: service-example
  annotations:
    konghq.com/plugins: rate-limiting-minute-10
spec:
  host: example.com
  controlPlaneRef:
    type: konnectNamespacedRef
    konnectNamespacedRef:
      name: cp
' | kubectl apply -f -
```

> At this point you can see the plugin is attached to the service in {{ site.konnect_short_name }}.

Reviewer: ```suggestion
At this point you can see the plugin is attached to the Service in {{ site.konnect_short_name }}.
```

You can also check the `KongPluginBinding` resource by running.

```shell
kubectl get kongpluginbinding
```

You can see the created `KongPluginBinding` like this:

```shell
NAME                            PLUGIN-KIND   PLUGIN-NAME                  PROGRAMMED
rate-limiting-minute-10-a0z1x   KongPlugin    rate-limiting-minute-10      True
```

### Attaching plugins to multiple entities

> {:.note}
> > **NOTE:** Binding plugins with this approach has limited observability and can yield unexpected results

Reviewer: ```suggestion
{:.info}
> **Note:** Binding plugins with this approach has limited observability and can yield unexpected results
```

> when multiple different resources are attached to the same plugin

> > (e.g. a service already has a plugin attached to it and we're annotating consumer with the same plugin).

Reviewer: ```suggestion
> (for example, a Service already has a plugin attached to it and we're annotating a Consumer with the same plugin).
```

> Users are advised to use `KongPluginBinding` CRD instead for better control and auditability.

Similar to those introduced above, you can also attach a plugin to multiple entities by configuring annotations of attached entities.
If a plugin appears in the `konghq.com/plugins` annotation of multiple entities, a `KongPluginBinding` will be created for the binding relationship between the plugin and the combination of these entities.

> Taking the example above where a plugin is attached to a service and a consumer:

Reviewer: ```suggestion
Taking the example above where a plugin is attached to a Service and a Consumer:
```
